pytest-embedded/pytest_embedded/plugin.py

```diff
@@ -1335,6 +1335,23 @@ def pytest_fixture_setup(self, fixturedef: FixtureDef[Any], request: SubRequest)
 
     @pytest.hookimpl(trylast=True)
     def pytest_collection_modifyitems(self, items: List[Function]):
+        def _check_duplicate_name(names_list: List[str], query: str):
+            if query == 'module name':
+                path_set = set(names_list)
+                names_list = [os.path.basename(name) for name in path_set]
+            if len(names_list) != len(set(names_list)):
+                for duplicate in set([name for name in names_list if names_list.count(name) > 1]):
+                    logging.error(
+                        f'A duplicate {query} name was discovered: {duplicate}. ' f'Each test {query} should be unique'
+                    )
+                exit(1)
+
+        case_name_list = [test.name for test in items]
+        case_module_list = [str(test.path.absolute()) for test in items]
+
+        _check_duplicate_name(case_name_list, query='case name')
+        _check_duplicate_name(case_module_list, query='module name')
```

Comment:
Sorry to point it out, but this kind of abstraction is actually making the code harder to read... you can actually put

```python
if query == 'module name':
    path_set = set(names_list)
    names_list = [os.path.basename(name) for name in path_set]
```

into the lines that make the actual call.

Reply:
Here I am not able to do that since variable `query` and the value of this variable doesn't have any source. This var is created by me

Reply:
```python
        def _check_duplicate_name(l: List[str]):
            if len(l) != len(set(l)):
                duplicates = set([name for name in l if l.count(name) > 1])
                if duplicates:
                    raise ValueError(f'Duplicated items: {duplicates}')
                exit(1)

        case_name_list = [test.name for test in items]
        case_module_list = [str(test.path.absolute()) for test in items]

        _check_duplicate_name(case_name_list)
        _check_duplicate_name([os.path.basename(name) for name in set(case_module_list)])
```
like this. Move the `query` logic into the calling statements, not the common function.

Reply:
even shorter, var `case_name_list` and `case_module_list` are not necessarily needed.